frontend/src/components/common/Resource/LogsButton.tsx

```diff
@@ -0,0 +1,425 @@
+import { Box, FormControl, InputLabel, MenuItem, Select, styled, Switch } from '@mui/material';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import { Terminal as XTerminal } from '@xterm/xterm';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { request } from '../../../lib/k8s/apiProxy';
+import { KubeContainerStatus } from '../../../lib/k8s/cluster';
+import Deployment from '../../../lib/k8s/deployment';
+import { KubeObject } from '../../../lib/k8s/KubeObject';
+import Pod from '../../../lib/k8s/pod';
+import ActionButton from '../ActionButton';
+import { LogViewer } from '../LogViewer';
+import { LightTooltip } from '../Tooltip';
+
+// Component props interface
+interface LogsButtonProps {
+  item: KubeObject | null;
+}
+
+// Styled component for consistent padding in form controls
+const PaddedFormControlLabel = styled(FormControlLabel)(({ theme }) => ({
+  margin: 0,
+  paddingTop: theme.spacing(2),
+  paddingRight: theme.spacing(2),
+}));
+
+export function LogsButton({ item }: LogsButtonProps) {
+  const [showLogs, setShowLogs] = useState(false);
+  const [pods, setPods] = useState<Pod[]>([]);
+  const [selectedPodIndex, setSelectedPodIndex] = useState<number | 'all'>('all');
+  const [selectedContainer, setSelectedContainer] = useState('');
+
+  const [logs, setLogs] = useState<{ logs: string[]; lastLineShown: number }>({
+    logs: [],
+    lastLineShown: -1,
+  });
+  const [allPodLogs, setAllPodLogs] = useState<{ [podName: string]: string[] }>({});
+
+  const [showTimestamps, setShowTimestamps] = useState<boolean>(true);
+  const [follow, setFollow] = useState<boolean>(true);
+  const [lines, setLines] = useState<number>(100);
+  const [showPrevious, setShowPrevious] = React.useState<boolean>(false);
+  const [showReconnectButton, setShowReconnectButton] = useState(false);
+
+  const xtermRef = React.useRef<XTerminal | null>(null);
+  const { t } = useTranslation(['glossary', 'translation']);
+
+  const clearLogs = React.useCallback(() => {
+    if (xtermRef.current) {
+      xtermRef.current.clear();
+    }
+    setLogs({ logs: [], lastLineShown: -1 });
+  }, []);
+
+  // Fetch related pods.
+  async function getRelatedPods(): Promise<Pod[]> {
+    if (item instanceof Deployment) {
+      const labelSelector = item.getMatchLabelsList().join(',');
+      const response = await request(
```

Comment:
There should be error handling for this. With a try/catch, and telling the user that something went wrong if something does.